### DISCUSSION TITLE
Install of pimcore AlternateObjectTreesBundle

### DISCUSSION THREAD

**Discussion by butlersupply** [pimcore/pimcore#7709] [Q&A / Support] [answered]
I've downloaded the AlternateObjectTreesBundle.  This is my first pimcore extension install and I have a few questions.

Do i copy the AlternateObjectTreesBundle directory into my /src directory, or do I copy it to the vendor/pimcore/pimcore/bundles directory.

And I guess i need to modify the namespace for all of the source files so that it will show up in the  extension manager?

Thanks!

**Comment by butlersupply** [comment, marked as answer]
Ok I think I have the path figured out.  I copied the source code to src/Elements/Bundle/AlternateObjectTreesBundle... and enabled the plugin.  But I am getting routing errors.  

Status: 404 | Not Found
URL: /admin/elements-alternate-object-trees/alternate-object-trees
Method: GET
Message: No route found for "GET /admin/elements-alternate-object-trees/alternate-object-trees" 

I've spent the afternoon looking at it but can't find the problem.  The route in the controller looks like this: 

@Route("/alternate-object-trees", methods={"GET"})

And the routing .yml has

_elements_alternate_object_trees:
    resource: "@ElementsAlternateObjectTreesBundle/Controller/"
    type:     annotation

**Reply by fashxp** [reply]
see also https://github.com/elements-at/AlternateObjectTrees/issues/20 ... can you please try this and provide a PR to alternateobjecttrees if it is working? 

thx...

**Reply by butlersupply** [reply]
Thank you.  That fixed the problem.  I'll give it a try and then review.

**Reply by butlersupply** [reply]
Sorry, I don't see where I can leave a PR.  On the pimcore site I would need to be a partner to review.

**Reply by dvesh3** [reply]
@butlersupply please create a pull request on https://github.com/pimcore/advanced-object-search 
follow guidelines:
 https://github.com/pimcore/pimcore/blob/master/CONTRIBUTING.md#contributing-pull-requests
thanks!
